src/services/services.ts

```diff
@@ -1011,6 +1014,22 @@ module ts {
         isWriteAccess: boolean;
     }
 
+    export interface DocumentHighlights {
+        fileName: string;
+        highlightSpans: HighlightSpan[];
+    }
+
+    export class DocumentHighlightKind {
+        public static definition = "definition";
+        public static reference = "reference";
+        public static writtenReference = "writtenReference";
+    } 
+
+    export interface HighlightSpan {
+        textSpan: TextSpan;

```

Comment:
That's a good point.  I'm trying to be 1:1 with the Roslyn API here.  They can't make HighlightSpan be a subtype of TextSpan as these are value types in the .net world and there is no value type subtypeing.